webapp/channels/src/components/channel_bookmarks/bookmark_item.tsx

```diff
@@ -88,18 +88,38 @@ const BookmarkItem = <T extends HTMLAnchorElement>({bookmark}: Props) => {
         );
     }
 
+    return {
+        link,
+        icon,
+        open,
+    } as const;
```

Comment:
Indeed, since the hook interface is an object, `as const` is redundant in this case. It's not for memoization, however I like being explicit about locking the type down. It's necessary to use `as const` for arrays as the type for each item becomes a union of all the other item types when used like I mention in the other comment.

Reply:
Ah, I see. That makes sense for when we're using arrays as tuples, but I'm not sure I like using it for cases like this object where it isn't needed since it kind of becomes noise that looks important when it isn't. That's sort of why I thought you might've been intending it to be used for memoization.

That's just an opinion though. 0/5 on actually changing it